lib/CGameState.cpp

```diff
@@ -740,6 +740,16 @@ void CGameState::init(const IMapService * mapService, StartInfo * si, bool allow
 
 	logGlobal->debug("Initialization:");
 
+	const JsonNode & baseBonuses = VLC->modh->settings.data["baseBonuses"];
+	logGlobal->debug("\tLoading global bonuses");
+	for(const auto & b : baseBonuses.Vector())
+	{
+		auto bonus = JsonUtils::parseBonus(b);
+		bonus->source = Bonus::GLOBAL;//for all
+		bonus->sid = -1; //there is one global object
+		globalEffects.addNewBonus(bonus);
+	}
+
 	initPlayerStates();
```

Comment:
Extracting in a new function would help a lot with code clarity.

Reply:
done